### PR TITLE
Added Date picker for filtering

### LINKED ISSUE #530: Fancy date filtering in tables
On the project table, I'd like it so filtering the start and end dates uses a datepicker.  Clicking on the a filter should bring up a datepicker where you can select a range of dates, and then the table will filter so that only that range is visible.  Will need to be easy to clear back to view all as well.

### PR DESCRIPTION
**Pull request by wongband**
resolve #530

### CODE REVIEW

**Review by jSylvestre — CHANGES_REQUESTED** [review]
Clearing the end date filter appears to hide the entire table contents.
Also, the end date filter should include dates that end on the date selected.

File: Harvest.Web/package-lock.json [hunk 1]
```diff
@@ -1,5 +1,13 @@
 {
   "name": "harvest.web",
   "version": "1.0.0",
-  "lockfileVersion": 1
+  "lockfileVersion": 2,
```
Comment (srkirkland) [inline Harvest.Web/package-lock.json:4]: Not sure this really makes sense, Harvest.Web doesn't have packages and shouldn't need a packages section in the package-lock.json file.

**Review by srkirkland — CHANGES_REQUESTED** [review]
I think you have a good start and the code that's there looks good.  But I don't think it makes a sense what is being filtered right now, what I was looking for in the original issue was that each filter would allow you to select a range of dates.  So for example, clicking on the start filter would bring up something where you could say show me everything that starts between October 1st and Nov 1st and then it would filter to that.

File: Harvest.Web/package-lock.json (hunk 1 above)
Comment (wongband) [reply Harvest.Web/package-lock.json:4]: hmmm strange, I don't remember adding that but I will remove it

**Comment by wongband** [comment]
@srkirkland, is there a specific way on how you want the filter to look like or could I just put two datepickers right under each other?

**Comment by srkirkland** [comment]
> @srkirkland, is there a specific way on how you want the filter to look like or could I just put two datepickers right under each other?

@wongband our datepicker has a built in method for specifying ranges.  see https://reactdatepicker.com/#example-date-range-for-one-datepicker

**Comment by srkirkland** [comment]
Or for your case, you probably want https://reactdatepicker.com/#example-date-range-using-input-with-clear-button

**Comment by wongband** [comment]
@srkirkland ooo awesome, I will take a look at it.

**Pull request closed without merging**